spec/controllers/application_controller_spec.rb

```diff
@@ -0,0 +1,19 @@
+require "rails_helper"
+
+RSpec.describe ApplicationController, type: :controller do
```

Comment:
As this is a user-facing feature, it would be better to have a user-centred spec. What I mean by this is that in general, it's good for our tests to describe what happens in terms of what the users experience, because that's a better way of "proving" that the site does what we expect. So we want our test to basically say:

> when the site is in maintenance mode, if I try to visit the start page I should see a message saying "Sorry there is a problem with the service"

The closest we can get to that in RSpec would be a feature spec (i.e. a new file `spec/features/maintenance_mode_spec.rb`, where we ideally switch on the feature flag in the same way we do for other tests (i.e. setting a tag like `:maintenance_mode_flag` in rails_helper.rb) that has a spec that says things like `visit root_path` and `expect(page).to have_text "Sorry there is a problem with the service")`

The spec you have here is fine, but it just checks that if maintenance mode is on, calling the `check_maintenance_mode` method will call the `redirect_to` method, and there could be some bugs that mean that, even though that is true, the end result of showing the correct error message to the user is not achieved.